dali/kernels/slice/slice_flip_normalize_permute_pad_gpu.h

```diff
@@ -165,8 +175,12 @@ class SliceFlipNormalizePermutePadGpu {
       for (int d = 0; d < nfill_values_; d++)
         fill_values[d] = sample_args.fill_values[d];
     }
-    OutputType *fill_values_gpu = context.scratchpad->ToGPU(
-        context.gpu.stream, make_span(fill_values_cpu, num_samples * nfill_values_));
+    OutputType *fill_values_gpu;
+    std::tie(norm_add_gpu, norm_mul_gpu, fill_values_gpu) = context.scratchpad->ToContiguousGPU(
```

Comment:
This is probably a long shot but what would be the difference if you allocate `norm_add_cpu`, `norm_mul_cpu`, and `fill_values_cpu` as one continuous block and then copy it as such (it should be just one copy invocation), but it would require manual pointer math.

Reply:
ToContiguousGPU will result in a single h2d copy. What we would have here are extra h2h copies to a pinned staging buffer. I think that we can live with this (very small amount of data copied). I wouldn't complicate this with manual pointer math.

What we could change is that the buffers for norm_add_cpu and norm_mul_cpu don't need to be pinned (since they are copied to a pinned staging buffer by ToContiguousGPU):

Example, we could do `AllocateHost` instead of `AllocatePinned`:
```
norm_add_cpu = context.scratchpad->AllocateHost<float>(num_samples * norm_args_size_);
```

We can always take care of this in a follow-up PR, as it is a little bit out of scope here.

Reply:
Agree